components/merkledb/src/db.rs

```diff
@@ -920,6 +920,61 @@ impl<'a> RawAccess for ReadonlyFork<'a> {
     }
 }
 
+/// Version of `ReadonlyFork` with a static lifetime. Can be produced from an `Rc<Fork>` using
+/// the `AsReadonly` trait.
+///
+/// Beware that producing an instance increases the reference counter of the underlying fork.
+/// If you need to obtain `Fork` from `Rc<Fork>` via [`Rc::try_unwrap`], make sure that all
+/// `ReadonlyRcFork` instances are dropped by this time.
+///
+/// [`Rc::try_unwrap`]: https://doc.rust-lang.org/std/rc/struct.Rc.html#method.try_unwrap
+///
+/// # Examples
+///
+/// ```
+/// # use exonum_merkledb::{access::AccessExt, AsReadonly, Database, ReadonlyRcFork, TemporaryDB};
+/// # use std::rc::Rc;
+/// let db = TemporaryDB::new();
+/// let fork = Rc::new(db.fork());
+/// fork.get_proof_list("list").extend(vec![1_u32, 2, 3]);
+/// let ro_fork: ReadonlyRcFork = fork.as_readonly();
+/// let list = ro_fork.get_proof_list::<_, u32>("list");
+/// assert_eq!(list.len(), 3);
+/// ```
+#[derive(Debug, Clone)]
+pub struct ReadonlyRcFork(Rc<Fork>);
```

Comment:
IMHO `Rc` in the middle looks a bit weird. I'd suggest `ReadonlyForkRc` (assuming that usually `Rc` is treated like a name, and not expanded mentally to "reference-counting").

Reply:
Well, I arrived to `ReadonlyRcFork` by mentally expanding `Rc`: "readonly [version of] reference-counted fork" 🙃 Maybe, something like `OwnedReadonlyFork` would make sense?

Reply:
Yep, I think so.